rust/ql/test/query-tests/unusedentities/main.rs

```diff
@@ -302,6 +350,27 @@ fn if_lets_matches() {
         p => { // BAD: unused variable
         }
     }
+
+    let duration1 = std::time::Duration::new(10, 0); // ten seconds
+    assert_eq!(duration1.as_secs(), 10);
+
+    let duration2:Result<std::time::Duration, String> =
+        Ok(std::time::Duration::new(10, 0));
+    match duration2 {
+        Ok(n) => { println!("duration was {} seconds", n.as_secs()); }
+        Err(_) => { println!("failed"); }
+    }
+
+    let (left,
+        right) = // BAD: unused value [NOT DETECTED] SPURIOUS: unused variable
```

Comment:
I think this should be `// BAD: unused variable`

Reply:
Makes sense. I like the second more, I would not consider a variable to be "used" just because it has an initializer.

Reply:
Perhaps we should just unify the two queries into a single one. The way they are currently defined, they should not be run in isolation anyway.  That would solve the subtle difference between unused value and variable.

Reply:
Hmm, I agree they should be used together but I kind of like the distinction.
- the distinct messages are helpful as they explain exactly what the complaint is in each case (mind you we could merge the queries but keep two messages).
- `rust/unused-value` is an SSA query, whereas `rust/unused-variable` is a near-trivial AST query.  Again, we can merge them anyway but it's not simplifying anything.
- currently they don't have the same `@precision`.  We actually can't do a lot about this.

Reply:
> This behaviour is actually the result of a single line in the `UnusedVariable.qll` library used by both queries.

I may have made it sound like the two queries have more in common than they do.  `UnusedVariable.qll` is the logic for the unused variable query.  The unused value query accesses it only to avoid producing "duplicate" results.